### PR TITLE
add misc setup docs and workflows

### PR DESCRIPTION
**Pull request by MSevey**
copied over various common workflows and initialization files from rollkit. 

with the new auto request workflow, we shouldn't need a CODEOWNERS file as that is covered in the auto request workflow. The auto request won't run until this PR is merged.

### CODE REVIEW

**Comment by MSevey** [comment]
NOTE: i expect most of the workflows to fail right now until the initial code is pushed as which point we can start fixing the workflows.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+coverage.txt
+proto/pb
+proto/tendermint
+types/pb/tendermint
```
Comment (nashqueue) [inline .gitignore:4]: no way to be gerneated 
```suggestion

```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	@$(MAKE) test-unit
+	@$(MAKE) test-unit-race
+.PHONY: test-all
+
+## proto-gen: Generate protobuf files. Requires docker.
+proto-gen:
+	@echo "--> Generating Protobuf files"
+	./proto/get_deps.sh
+	./proto/gen.sh
+.PHONY: proto-gen
```
Comment (nashqueue) [inline Makefile:57]: scripts do not exist fro generation
```suggestion

```

**Review by nashqueue — COMMENTED** [review]
TY :) small nits on proto generation

**Review by nashqueue — DISMISSED** [review]

**Comment by nashqueue** [comment]
We are checking lints against Go version 1.20, which might have a conflict with rollkit.

**Review by tzdybal — COMMENTED** [review]
After rebasing this should be ready for merging.

**Review by nashqueue — APPROVED** [review]
LGTM

**Review by tzdybal — APPROVED** [review]

**Pull request merged**